src/core/qgsdistancearea.cpp

```diff
@@ -537,6 +540,111 @@ double QgsDistanceArea::measureLine( const QgsPoint &p1, const QgsPoint &p2, Qgs
   return result;
 }
 
+double QgsDistanceArea::measureLineProjected( const QgsPoint &p1, double distance, double azimuth, QgsPoint *projected_point )
+{
+  double result = 0.0;
+  QgsPoint p2;
+  if ( geographic() )
+  {
+    if ( !mEllipsoidalMode )
+    {
+      if ( ( mSemiMajor > 0 ) && ( mSemiMinor > 0 ) )
+      {
+        setEllipsoid( mSemiMajor, mSemiMinor );
+      }
+      else
+      {
+        setEllipsoid( sourceCrs().ellipsoidAcronym() );
+      }
+      if ( mEllipsoid != GEO_NONE )
+      {
+        mEllipsoidalMode = true;
+      }
+    }
+    if ( mEllipsoid != GEO_NONE )
+    {
+      p2 = computeSpheroidProject( p1, distance, azimuth );
+      QList<QgsPoint> linePoints;
```

Comment:
This can be optimised - use `result = p1.distance(p2)` instead and avoid the creation of the allocating lists & linestring conversion.